juju/juju
# S3Client PUT and DELETE methods

Adds PUT and DELETE methods to the s3caller client. The s3caller client is a wrapper around the s3client to ensure it works with the dependency engine.

The code adds simple PutObject and DeleteObject methods to the s3client. A new credentials type allows us to swap the credentials based on what the caller wants. This will be interesting to manage, as you'll likely want a s3client with different credentials.

For now, I've baked in the anonymous credentials and we can iterate on that.

Lastly, we should rip out the httprequest.Client from the API package. We shouldn't be forcing consumers of the API client to use some poorly maintained http.Client wrapper.





## Checklist




## QA steps

Regression test to ensure we didn't break anything.

```sh
$ juju bootstrap lxd test --build-agent
$ juju deploy ubuntu
```

## Links


**Jira card:** JUJU-5291
